app/src/main/java/org/mozilla/fenix/onboarding/view/UpgradeOnboardingScreen.kt

```diff
@@ -0,0 +1,215 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.onboarding.view
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material.Icon
+import androidx.compose.material.IconButton
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.google.accompanist.insets.navigationBarsPadding
+import com.google.accompanist.insets.statusBarsPadding
+import mozilla.telemetry.glean.private.NoExtras
+import org.mozilla.fenix.GleanMetrics.Onboarding
+import org.mozilla.fenix.R
+import org.mozilla.fenix.theme.FirefoxTheme
+
+/**
+ * Enum that represents the onboarding page that is displayed.
+ */
+private enum class UpgradeOnboardingState {
+    Welcome,
+    SyncSignIn,
+}
```

Comment:
Any thoughts on combining `OnboardingPageUiState` and `UpgradeOnboardingState` or making `UpgradeOnboardingState` a child class? It could simplify the `OnboardingPage(...)` initialization with the giant `when`.

Reply:
My understanding was that this work was motivated by https://www.figma.com/file/d7l1iGabQYqnLcnQbHDGzM/Wallpaper-2.0-%2B-Onboarding-v106?node-id=8924%3A220946&t=ihw1RVyD5zhJ7Gze-4 which is to add an onboarding screen for push notifications. Therefore, keeping `OnboardingPageUiState` might still make sense.

Reply:
I have also briefly discussed with @rahulsainani about the pros and cons with the status quo with managing this `enum` state vs adopting a `NavHost` to get rid of the `when`. At the end of the day, both approaches would be  fine. I have filed https://github.com/mozilla-mobile/fenix/issues/28330 last week with the intent to move this to use navigation to remove the `when` that is going on.

Rahul have said "there's not a lot of navigation happening and we'd have to clear the back stack anyway as we don't allow users to go back on Welcome screen. It's fine either way imo". I am mostly okay with keeping the status quo since the navigation is minimal and not having to handle the back stack. We can keep https://github.com/mozilla-mobile/fenix/issues/28330 around for future exploration if we felt strongly about it.

Reply:
Gotcha gotcha. I'm definitely open to there being a navigation refactor down the road, and I'm glad yall had that talk to figure out what makes sense. I'm open to moving forward as is, with the caution if onboarding, for some reason, gets interest in becoming more complex.

Reply:
@MozillaNoah were you suggesting something like this?

```
enum class OnboardingPageUiState(
    @DrawableRes val image: Int,
    @StringRes val title: Int,
    @StringRes val description: Int,
    @StringRes val primaryButtonText: Int,
    @StringRes val secondaryButtonText: Int? = null,
    val recordImpressionEvent: () -> Unit,
) {
    Welcome(
        image = ...
    ),
    SyncSignIn(
        image = ...
    )
}
```

If so, then I'd +1 it, as I think it well encapsulates the static differences between the pages well. I do think we will need to move the metrics callbacks into a a `Controller` class and passed to the Composables per gl's comment below in order to get previews rendering.

Reply:
@MatthewTighe This is also an interesting idea.

Reply:
The idea was that `OnboardingPage` is in a different component and doesn't need to who/what uses it, similar to a `Button`, the callers can keep the objects and pass on the data, as there might be more features using the OnboardingPage. Also makes it possible to move `OnboardingPage` to a different module, let me know your thoughts

Reply:
That is a good point. If we consider this with the possibility of modularizing Fenix in the future I think our options are:

1. Treat onboarding as a complete feature vertical, in which case my suggestion above would still work. Other features that required onboarding could come add their static properties in the onboarding module itself.
2. Define onboarding pages per feature module and pass them to the onboarding module, as it sounds like you're suggesting here. If we went this route, another convention you could consider is to define a top-level `val` for each onboarding page. For example, in `/feature/onboarding/Onboarding.kt` we could have `val welcomeOnboarding = OnboardingPageUiState(...)`.

I don't necessarily have a strong opinion here, but I would lean towards the 1st option still. My impression when onboarding is discussed here is that we will be treating it as a distinct feature in terms of additions, experimentation, etc. I think my example in 2 also illustrates what could become a slightly awkward module structure - it would make sense to have something like `/feature/sync/Onboarding` or `/feature/notifications/Onboarding`, but then those instances are split from any number of pages that could end up defined in `/feature/onboarding/Onboarding`.

It also really doesn't matter for now, I don't think, as this is all hypothetical work. Just wanted to share some of my thoughts. Plans for the future never survive first contact anyway, so I'd stick with whatever feels most comfortable to you 🙂

Reply:
@MatthewTighe, yes exactly that!

Reply:
@MatthewTighe thanks for the explanation and the thought process. I think that's a bit outside the scope of this PR. Also, Welcome and Sync Sign In Onboarding uses Indicators (which are not touched) and the upcoming notification pre permission dialog doesn't use it, that's why the smallest possible component (OnboardingPage) was extracted so we can reuse that part. I'm open to commonalising more parts of the UI but I would say we should do that going forward as things get more clear based on the usages. wdyt?